BookingV2.Logic/Services/RealEstateService.cs

```diff
@@ -0,0 +1,167 @@
+using BookingV2.Logic.Contract;
+using BookingV2.Logic.Responses;
+using BookinV2.Data;
+using BookinV2.Data.Entities.RealEstateEntities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace BookingV2.Logic.Services
+{
+    public class RealEstateService : IRealEstateService
+    {
+        private readonly BookingV2DBContext _context;
+        private readonly ILogger<RealEstateService> _logger;
+
+        public RealEstateService(BookingV2DBContext context, ILogger<RealEstateService> logger)
+        {
+            this._context = context ?? throw new ArgumentNullException(nameof(context));
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task<ApiResponse<IEnumerable<RealEstate>>> GetAllAsync()
+        {
+            var response = new ApiResponse<IEnumerable<RealEstate>>();
+
+            try
+            {
+                if (this._context.RealEstates == null)
```

Comment:
it is kinda impossible situation when real dbSet is null. When you change type to our abstraction this check become useless